x-pack/plugins/security_solution/common/experimental_features.ts

```diff
@@ -105,6 +105,11 @@ export const allowedExperimentalValues = Object.freeze({
    * */
   discoverInTimeline: false,
 
+  /**
+   * disables ES|QL rules
+   */
+  esqlRulesDisabled: false,
```

Comment:
Could we change this flag to a positive name?
```suggestion
  esqlRulesEnabled: true,
```

Reply:
@semd 

Esql is planned to be available by default, only disabled for Serverless. That's why feature flag with negative name, to disable this rule type.

Also, if I understand correctly, once feature is set `true` here, it cannot be unset to `false` - what we need for Esql.

Reply:
@vitaliidm I understand what you mean, but this part in the serverless config:
```
xpack.securitySolution.enableExperimental:
  - esqlRulesDisabled
```
Enables the experimental flag in order to disable the feature, which is very confusing. 

From what you said, this feature is not actually experimental, we just want to disable it for serverless. That's not what experimental flags are intended for, so we should not be using it this way, there are better ways to do that:

- Create a separate config flag (e.g. `xpack.securitySolution.enableEsqlRules`) that defaults to `true`, and set it to `false` in serverless.

- Expose a flag (setter) in the `security_solution` plugin contract. and then set it to _true_ in `security_solution_ess` and _false_ in `security_solution_serverless` plugins respectively.

Would it be possible to apply any of those solutions?

Reply:
@semd , thanks for outlining 2 possible options

I have implemented one with config, similar to the other configs for serverless: https://github.com/elastic/kibana/pull/165450/commits/7ac61a77bc8a1bf7161b79ccd9ecb2d7b84d1976

That approach required though much more code changes, especially around pulling config values to UI and work with them, required to update interface of KibanaService, which for some flag, intended to be removed in future looks like overkill. Since this would require rolling back interface once flag not there anymore. Also, as I understand, that config value would leave forever in kIbana, since removing it would be breaking change for Kibana config.

I still think, that experimental feature flag, can be suitable in this case, since it's allows easily plug and play with config values on UI and server, and disabling something thought it, while might be deemed as not be conventional use but does its job.

Other thought, while I worked on adding `xpack.securitySolution.enableEsqlRuleType` , it seems too complicated to use that config value in UI(see changes to KibanaService interface). Am I missing something or should we come up in future with the simpler way to accessing config values in UI?

Reply:
we had a discussion with @elastic/security-detection-engine regarding the way we would like to configure esql rule availability. And considering, that we do not plan to have this option indefinitely(it would hide it temporary, until esql will be released on serverless) - creation of a separate, non-removable in future config option, is not the way we would like to proceed

Reply:
@vitaliidm Okay, considering this is temporary, I am fine using the experimental flags in this "non-conventional" way.

Actually, this makes me think it would be great to have an `ess.security.yml` that would be parsed only in non-serverless environments, this way we could enable this kind of experimental flag normally, without using tricky double negations on the serverless side.

Anyway, it LGTM 👍

Reply:
thanks!

> Actually, this makes me think it would be great to have an ess.security.yml that would be parsed only in non-serverless environments, this way we could enable this kind of experimental flag normally, without using tricky double negations on the serverless side.

agree, having similar config for ess would be helpful